apis/v1/gateway_types.go

```diff
@@ -297,7 +297,7 @@ type AllowedListeners struct {
 	// While this feature is experimental, the default value is to allow no ListenerSets.
 	//
 	// +optional
-	// +kubebuilder:default={from: None}
+	// +kubebuilder:default={from: Same}
```

Comment:
I think it's still useful to make ListenerSet opt-in by default, so I personally prefer the previous `None` default.

Reply:
Yeah, agreed. I think also that changing from `None` -> `Same` counts as loosening validation, so we could conceivably do this later too, if we wanted.